examples/advanced/pi-lluminate.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Use the Enviro pHAT to check how much light is in a room, then trigger a webhook to control
+# Philips Hue lights via IFTTT. By Wesley Archer (@raspberrycoulis).
+
+# Import the relevant modules
+
+import requests
+import time
+import datetime
+import os
+from envirophat import light
+
+# Get the current time for displaying in the terminal.
+def whats_the_time():
+    now = datetime.datetime.now()
+    return (now.strftime("%H:%M:%S"))
+
+# The function to turn off the lights. Sends a webhook to IFTTT which
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_off():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights off!")
+
+# The function to turn on the lights. Sends a webhook to IFTTT which                   
+# triggers Hue. Replace the trigger word and tokens from your account.
+
+def turn_on():
+    requests.post("https://maker.ifttt.com/trigger/{TRIGGER_WORD}/with/key/{TOKEN_GOES_HERE}")
+    print("Lights on!")
+
+# Check the light level and determine whether the lights need to 
+# be turned on or off.
+
+def average_lux():
+    # Variables for calculating the average lux levels
+    start_time = time.time()
+    curr_time = time.time()
+    collect_light_time = 60
+    collect_light_data = []
+
+    # Calculate the average lux level over 60 seconds
+    print("Calculating average light level...")
+    while curr_time - start_time < collect_light_time:
+	      curr_time = time.time()
+	      avg = light.light()
+	      collect_light_data.append(avg)
+	  time.sleep(1)
+    average_light = sum(collect_light_data[-10:]) / 10.0
+    now = whats_the_time()
+    print("{} {} {} {} {} {} {} {}.".format("Average over", collect_light_time, "seconds", "is:", average_light, "lux.", "Last checked at", now))
+    print("{} {} {} {}.".format("Waiting", "90", "seconds", "before trying again"))
+    return average_light
+
+try:
+    # Local variables.
+    state = 0	# Sets the state for the lights.
+    low = 260	# Low value for light level (lux).
+    high = 300	# High value for light level (lux).
+    period = 90	# Delay, in seconds, between calls.
+    while True:
+	# Get the average lux level first,
+	room_light = average_lux()
+	# Now check if the room is dark enough then turn on the lights.
+	if room_light < low and state != 1:
```

Comment:
You should use `state = True` and `state = False` and simplify this to:

```python
if room_light < low and not state:
```

and:

```python
elif room_light > high and state:
```

Technically 1 and 0 are totally interchangeable with True/False, but they do not clearly communicate the boolean state of your system so it would be clearer to use True/False.

In this instance I'd probably change `state` to `lights_on` so that your code is much easier for a human to parse:

```python
if room_light < low and not lights_on:
    turn_on()
    lights_on = True
elif room_light > high and lights_on:
    turn_off()
    lights_on = False
```

Reply:
Thanks, that makes sense. To be honest, I based that part on the guide Sandy published around controlling the [IKEA Trådfri Lights](https://learn.pimoroni.com/tutorial/sandyj/controlling-ikea-tradfri-lights-from-your-pi) which uses `1` and `0`.

Reply:
Thanks @Gadgetoid, appreciate the feedback and suggestions. I'll review these and come back with an update.